src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -682,32 +755,54 @@ private void fillFlightContexts(Connection connection, List<FlightContext> fligh
   }
 
   /**
-   * Return flight state for a single flight
+   * Return flight state for a single flight. The status of completed flight is limited by
+   * completeFlightAvailable property.
    *
    * @param flightId flight to get
    * @return FlightState for the flight
+   * @throws FlightNotFound - no flight
+   * @throws DatabaseOperationException - unexpected error in the operation
+   * @throws InterruptedException - interrupt
    */
   FlightState getFlightState(String flightId)
       throws DatabaseOperationException, InterruptedException {
-    return DbRetry.retry("flight.getFlightState", () -> getFlightStateInner(flightId));
+    return DbRetry.retry("flight.getFlightState", () -> getFlightStateInner(flightId, true));
+  }
+
+  FlightState controlGetFlightState(String flightId)
```

Comment:
Worth a comment on how this is different from `getFlightState`. `control` doesn't really explain much to me on its own.